array-manipulation-2/src/chunk.js

```diff
@@ -0,0 +1,29 @@
+/* exported chunk */
+const chunk = (array, size) => {
+  if (size >= array.length) {
+    return array;
+  } else {
+    // const arrCopy=[...array];
+    const chunkNumber = Math.floor(array.length / size);
+    const remainingNumber = array.length % size;
+    const result = [];
+    const chunk = [];
+    let startingIndex = 0;
+    for (let i = 0; i < chunkNumber; i++) {
+      for (let j = 0; j < size; j++) {
+        chunk.push(array[startingIndex + j]);
+      }
+      const chunkCopy = [...chunk]; // without this copy, everytime chunk resets, the result becomes empty
+      result.push(chunkCopy);
+      startingIndex += size;
+      chunk.length = 0; // reset chunk array for next loop.
+    }
+    if (remainingNumber !== 0) {
+      const remainingArray = array.slice(-remainingNumber); // may need more loops if not using slice
+      result.push(remainingArray);
+      return result;
+    } else {
+      return result;
+    }
+  }
+};
```

Comment:
We can actually simplify this code a lot! Check out our solution
```
function chunk(array, size) {
  const chunks = [];
  for (let i = 0; i < array.length; i++) {
    if (i % size === 0) {
      chunks.push([array[i]]);
    } else {
      const lastChunk = chunks[chunks.length - 1];
      lastChunk.push(array[i]);
    }
  }
  return chunks;
}
```

Reply:
Thank you Shawn, I understand my code is clumsy. I will try to keep it simple and refactor codes next time.

Reply:
No need to apologize Jimmy! I just wanted to show you another way of accomplishing this!